### DISCUSSION TITLE
Unexpected warning hint when coding with pandas.DataFrame.loc

### DISCUSSION THREAD

**Discussion by dangjinghao** [microsoft/pyright#4693] [Q&A] [answered]
Hello, I'm not very familiar with python or pylance.I will try to describe it in detail. 

I'm learning pandas. When I code `df.loc[len(df)]=["foo4","value4"] `, the pylance warned a message:
```
Argument of type "int" cannot be assigned to parameter "idx" of type "tuple[_IndexSliceTuple[Unknown], HashableT@__setitem__]" in function "__setitem__"
  "int" is incompatible with "tuple[_IndexSliceTuple[Unknown], HashableT@__setitem__]"
```
But this code could be run by `python test.py` or python interactive mode. In fact that it is often used as appending row.

Now I just could suppress the warning by adding ignore comment.

Example code
```python
import pandas as pd
foo_data = [["foo1","value1"],["foo2","value2"],["foo3","value3"]]
df=pd.DataFrame(data=foo_data)
df.loc[len(df)]=["foo4","value4"] 
df

```

some infomation:
```
ssh remote development

conda  4.12.0
numpy                     1.24.2           py39h7360e5f_0    https://mirrors.ustc.edu.cn/anaconda/cloud/conda-forge
pandas                    1.5.3            py39h2ad29b5_0    https://mirrors.ustc.edu.cn/anaconda/cloud/conda-forge
vscode 1.7.5.1
python 3.9.15


名称: Pylance
ID: ms-python.vscode-pylance
说明: A performant, feature-rich language server for Python in VS Code
版本: 2023.2.40
发布者: Microsoft
VS Marketplace 链接: https://marketplace.visualstudio.com/items?itemName=ms-python.vscode-pylance

```
![image](https://user-images.githubusercontent.com/53005337/221354471-d0a5f857-96eb-4dab-9614-92925ccc1272.png)

What happened with it and what should I do?
Thank you for your help!

**Comment by erictraut** [comment]
Pyright is working as designed here, so this isn't a pyright bug. I don't know enough about pandas to say whether this is a bug in your code or a bug in the type stubs that are bundled with pylance. In either case, the pylance team is probably best able to answer your question since they maintain the bundled pandas stubs.

@gramster, any thoughts on this one?

**Reply by twoertwein** [reply, marked as answer]
I believe that might be a bug in pandas-stubs. @dangjinghao feel free to open an issue here https://github.com/pandas-dev/pandas-stubs

I think `DataFrame.loc[value_not_existing_in_index] = new_row` is not recommended. You might want to use at [pd.concat](https://pandas.pydata.org/docs/reference/api/pandas.concat.html) to append rows.

**Reply by dangjinghao** [reply]
Thank you so much. This warning is caused by incomplete  pandas stubs and my fix PR has been merged. You can see [this PR](https://github.com/pandas-dev/pandas-stubs/pull/816)
